### ISSUE TITLE
CA2009 crashes on `new int[0].ToImmutableArray()` (.NET 8 Preview 1)

### ISSUE THREAD

**Issue by alexrp** [dotnet/roslyn-analyzers#6501]
### Analyzer

**Diagnostic ID**: [CA2009](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2009): `Do not call ToImmutableCollection on an ImmutableCollection value`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.100-preview.1.23115.2](https://dotnet.microsoft.com/download/dotnet/8.0)

### Describe the bug

CA2009 is crashing when calling `ToImmutableArray()` on an array creation expression.

### Steps To Reproduce

```
$ cat tests.csproj
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <AnalysisLevel>latest-all</AnalysisLevel>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>
</Project>
$ cat Program.cs
using System.Collections.Immutable;

var values = new int[0].ToImmutableArray();
```

### Expected behavior

No exception.

### Actual behavior

`CSC : warning AD0001: Analyzer 'Microsoft.NetCore.Analyzers.ImmutableCollections.DoNotCallToImmutableCollectionOnAnImmutableCollectionValue
Analyzer' threw an exception of type 'System.InvalidCastException' with message 'Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel.ArrayTypeSymbol' to type 'Microsoft.CodeAnalysis.INamedTypeSymbol'.'.`

**Comment by alexrp** [comment]
@Evangelink can this be closed as of #6476?

**Comment by alexrp** [comment]
Seems to be fixed.
